### PR TITLE
refactor: make trader support multiple networks

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-2425

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-state/src/Others.ts [hunk 1]
```diff
@@ -102,4 +106,17 @@ export class OthersState<ChainId, SchemaType, ProviderType, NetworkType, Transac
     formatSchemaType(schema: SchemaType): string {
         throw new Error('Method not implemented.')
     }
+    createNativeToken(chainId: ChainId): FungibleToken<ChainId, SchemaType> {
+        throw new Error('Method not implemented.')
+    }
+    createERC20Token(
```
Comment (guanbinrui) [inline packages/web3-state/src/Others.ts:112]:
```suggestion
    createFungibleToken(
```

File: packages/web3-state/src/Others.ts [hunk 2]
```diff
@@ -102,4 +106,17 @@ export class OthersState<ChainId, SchemaType, ProviderType, NetworkType, Transac
...
+        chainId: ChainId,
+        address: string,
+        name?: string,
+        symbol?: string,
+        decimals?: number,
+        logoURI?: string,
+    ): FungibleToken<ChainId, SchemaType> {
```
Comment (guanbinrui) [inline packages/web3-state/src/Others.ts:119]:
```suggestion
    createFungibleToken(
        chainId: ChainId,
        address: string,
        schemaType: SchemaType,
        name?: string,
        symbol?: string,
        decimals?: number,
        logoURI?: string,
    ): FungibleToken<ChainId, SchemaType> {
```

Comment (guanbinrui) [inline packages/web3-state/src/Others.ts]:
```suggestion
    createFungibleToken(
        chainId: ChainId,
        address: string,
        schemaType: SchemaType,
        name?: string,
        symbol?: string,
        decimals?: number,
        logoURI?: string,
    ): FungibleToken<ChainId, SchemaType> {
```

Alongside this method as well add `createNonFungibleToken`.

File: packages/web3-shared/base/src/specs/index.ts [hunk 3]
```diff
@@ -1421,6 +1422,10 @@ export interface OthersState<ChainId, SchemaType, ProviderType, NetworkType, Tra
     getMaskTokenAddress(chainId?: ChainId): string | undefined
     getAverageBlockDelay(chainId?: ChainId, scale?: number): number
     getTransactionSignature(chainId?: ChainId, transaction?: Partial<Transaction>): string | undefined
+
+    // #region Constructor
+    createNativeToken(chainId: ChainId): FungibleToken<ChainId, SchemaType>
+    createERC20Token( chainId: ChainId, address: string, name?: string, symbol?: string, decimals?: number, logoURI?: string): FungibleToken<ChainId, SchemaType>
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    createFungibleToken( chainId: ChainId, address: string, schemaType: SchemaType, name?: string, symbol?: string, decimals?: number, logoURI?: string): FungibleToken<ChainId, SchemaType>
    createNonFungibleToken( chainId: ChainId, address: string, schemaType: SchemaType, name?: string, symbol?: string, decimals?: number, logoURI?: string): FungibleToken<ChainId, SchemaType>
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/InputTokenPanel.tsx [hunk 4]
```diff
@@ -1,29 +1,26 @@
...
 import type { SelectTokenChipProps } from '@masknet/shared'
+import type { Web3Helper } from '@masknet/web3-helpers'
 
 export interface InputTokenPanelProps extends withClasses<'root'> {
     balance: string
     amount: string
     chainId: ChainId
     maxAmount: string
-    token?: FungibleToken<ChainId, SchemaType> | null
+    token?: FungibleToken<Web3Helper.ChainIdAll, Web3Helper.SchemaTypeAll> | null
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/InputTokenPanel.tsx]: Type `FungibleToken<Web3Helper.ChainIdAll, Web3Helper.SchemaTypeAll>` is using a lot. Please add `FungibleTokenAll` and `NonFungibleTokenAll` in `@masknet/helpers`.

File: packages/web3-shared/base/src/specs/index.ts [hunk 5]
```diff
@@ -1421,6 +1422,11 @@ export interface OthersState<ChainId, SchemaType, ProviderType, NetworkType, Tra
     getMaskTokenAddress(chainId?: ChainId): string | undefined
     getAverageBlockDelay(chainId?: ChainId, scale?: number): number
     getTransactionSignature(chainId?: ChainId, transaction?: Partial<Transaction>): string | undefined
+
+    // #region Constructor
+    createNativeToken(chainId: ChainId): FungibleToken<ChainId, SchemaType>
+    createFungibleToken(chainId: ChainId, schemaType: SchemaType, address: string, name?: string, symbol?: string, decimals?: number, logoURI?: string): FungibleToken<ChainId, SchemaType>
+    createNonFungibleToken(chainId: ChainId, address: string, schemaType: SchemaType, tokenId: string, ownerId?: string, metadata?: NonFungibleToken<ChainId, SchemaType>['metadata'], contract?: NonFungibleToken<ChainId, SchemaType>['contract'], collection?: NonFungibleToken<ChainId, SchemaType>['collection']): NonFungibleToken<ChainId, SchemaType>
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts:1437]:
```suggestion
    // #region token constructor
    createNativeToken(chainId: ChainId): FungibleToken<ChainId, SchemaType>
    createFungibleToken(chainId: ChainId, schemaType: SchemaType, address: string, name?: string, symbol?: string, decimals?: number, logoURI?: string): FungibleToken<ChainId, SchemaType>
    createNonFungibleToken(chainId: ChainId, address: string, schemaType: SchemaType, tokenId: string, ownerId?: string, metadata?: NonFungibleToken<ChainId, SchemaType>['metadata'], contract?: NonFungibleToken<ChainId, SchemaType>['contract'], collection?: NonFungibleToken<ChainId, SchemaType>['collection']): NonFungibleToken<ChainId, SchemaType>
    // #endregion
```

Comment (guanbinrui) [reply packages/web3-shared/base/src/specs/index.ts:1437]: @zhouhanseng will replace this in #7853.

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/TraderDialog.tsx [hunk 6]
```diff
@@ -70,47 +78,59 @@ const useStyles = makeStyles()((theme) => ({
 export function TraderDialog() {
```
Comment (nuanyang233) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/TraderDialog.tsx:78]: I removed the `TraderProps` state and added `defaultCoins`. cc @zhouhanseng

Comment (hanszhoou) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trader/TraderDialog.tsx:78]: LGTM.

File: packages/mask/src/plugins/Trader/trader/uniswap/usePools.ts [hunk 7]
```diff
@@ -54,22 +56,25 @@ export function usePools(
         }
     }, [chainId, transformed, context?.FACTORY_CONTRACT_ADDRESS])
 
-    const poolContracts = usePoolContracts(chainId, poolAddresses)
+    const poolContracts = usePoolContracts(
+        pluginID !== NetworkPluginID.PLUGIN_EVM ? (chainId as ChainId) : undefined,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/uniswap/usePools.ts:60]: Did you mean...?

```suggestion
        pluginID === NetworkPluginID.PLUGIN_EVM ? (chainId as ChainId) : undefined,
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**